### DISCUSSION TITLE
Jakarta Servlet 6.x specific features used in GraphQL Java Servlet

### DISCUSSION THREAD

**Discussion by ideas-into-software** [graphql-java-kickstart/graphql-java-servlet#508] [Q&A] [answered]
I was recently tasked with upgrading our open source project, Gecko GraphQL, from version 7.2.0 to 15.0 of GraphQL Java Servlet. This was the first version available, AFAIK, which consumed Jakarta Servlet instead of Javax Servlet.

We utilize EMF heavily throughout our open source and commercial projects and Gecko GraphQL so far allowed to use GraphQL with EMF. We also use OSGi heavily, and that includes Apache Felix.

Latest versions of both Apache Felix HTTP Jetty (`org.apache.felix.http.jetty`) Apache Felix HTTP Bridge (`org.apache.felix.http.bridge`) support Jakarta Servlet 5.x but exclude version 6.x or above, while GraphQL Java Servlet (`graphql-java-servlet`) supports Jakarta Servlet starting at version 6.x. 

Therefore, I wanted to ask are there any Jakarta Servlet 6.x specific features used in GraphQL Java Servlet ? If possibly there aren’t any, we could relax the version restrictions and be able to use GraphQL Java Servlet along latest available version of Apache Felix HTTP Jetty / Bridge.

**Comment by federicorispo** [comment, marked as answer]
@ideas-into-software In the last release the version of `springframework-*` dependencies were bumped to `6.0.5` and all the codebase was migrated from javax to jakarta. Unfortunately spring version 6.x requires jakarta 6.x (see [here](https://github.com/spring-projects/spring-framework/wiki/What%27s-New-in-Spring-Framework-6.x#testing)) so the jakarta downgrade is not feasable. However the graphql-java-servlet uses the springframework dependencies only for the unit tests and I think it could be useful to have a branch of this project that does not necessarily depend on the springframework ecosystem.

I'm already working to get more flavors of this library supporting jdk11 and javax (see this [thread](https://github.com/graphql-java-kickstart/graphql-java-servlet/discussions/487#discussioncomment-4949727)) and I'm happy to help maintaing also this flavor removing the spring dependencies but I'm stuck because the author hasn't created the branches where I can open PR.

I also bumped the graphql-java dependencies to 20.0.0 via [PR](https://github.com/graphql-java-kickstart/graphql-java-servlet/pull/503) but it seems like there is not much responsiveness in this project

**Comment by stbischof** [comment]
i highly recomment to use jakarta 5 
https://jakarta.ee/specifications/servlet/5.0/

since only a MilestoneRelease(M4) of glassfish  passes the tck. see https://jakarta.ee/specifications/servlet/6.0/

**Comment by ideas-into-software** [comment]
@federicorispo Thank you very much for your reply and tip. This is now taken care and I shared this via pull request: https://github.com/graphql-java-kickstart/graphql-java-servlet/pull/524 Regards
